ZUR-155 My PR adds a feature

tech-911 opened the pull request:
![image](https://user-images.githubusercontent.com/61203357/201394197-ab7cded5-496b-4ca1-b4f3-b6b234532d4c.png)

Added UI for changing the profile picture. ZUR-155. 
created a user profile picture upload UI, removed unused components, removed console logs, renamed image from pp to profileAvatar which is a more specific name

JulianaSau commented:
@tech-911 your pr title should be more descriptive. 
Also include a description of the changes that you made since you changed six files

JulianaSau commented on packages/main/src/components/protected/topbar/components/ProfilePicView.jsx:
```diff
@@ -1,12 +1,15 @@
 const ProfilePicView = () => {
   const { profilePicView, setProfilePicView } = useContext(TopbarContext);
   const { userProfileImage, user } = useContext(ProfileContext);
+  console.log(profilePicView);
```
It is not good practice to push console.log to a codebase. Ensure you've removed it from your commits

tech-911 replied:
ive resolved this

JulianaSau commented on packages/main/src/components/protected/topbar/components/ProfilePicView.jsx:
```diff
@@ -1,12 +1,15 @@
 import React, { useContext } from "react";
-import { TopbarContext } from "../context/Topbar";
-import { ProfileContext } from "../context/ProfileModal";
+// import { TopbarContext } from "../context/Topbar";
```
If the code is not being used, remove it entirely instead of commenting it out

tech-911 replied:
ive resolved this

JulianaSau commented on packages/main/src/components/top-navigation-bar/TopNavigationBar.jsx:
```diff
@@ -18,6 +18,10 @@ import ar from "./assets/language/ar.png";
 import he from "./assets/language/he.png";
 import pt from "./assets/language/pt.png";
 import world from "./assets/language/world.png";
+import pp from "./assets/language/pp.png";
```
Give this image a more descriptive name. `pp` could mean anything

tech-911 replied:
ive resolved this

JulianaSau commented:
Also, I think you misunderstood the task entirely. You were supposed to add an interface that enables changing the profile picture. What you added is just an image.
I believe this might be it
![profile](https://user-images.githubusercontent.com/49183775/201433360-e5c5baa4-a75f-4dae-b027-5072e3480dae.png)

tech-911 commented:
good eveing ma. pls ma i need more clarification of the task this is what i was given on linear: 
![image](https://user-images.githubusercontent.com/61203357/201437682-e7b4740c-d946-41ca-8a99-e8439d6aac00.png)

i dont understand what im to implement again. i thought it was just ui for profile upload. and also when i used the componet that has already been built i discovered it has issues and is giving error so am i to create my own component that when i click the profile icon this appears: 
https://user-images.githubusercontent.com/49183775/201433360-e5c5baa4-a75f-4dae-b027-5072e3480dae.png

and then what else should i implement

JulianaSau commented:
You are supposed to create the modal that will be used to edit the user's profile picture. The modal i posted above. So your focus shouls be on th eright side where there's `upload photo` and `remove photo`

tech-911 commented:
ok and also should i add the save changes and cancel button also should it be the dark theme as here too

tech-911 commented:
pls ma is this ok :

![image](https://user-images.githubusercontent.com/61203357/201441218-79fe1407-039c-4747-9e7e-aa3a5672e1cf.png)

JulianaSau commented:
@billmal071 help me review this one. Will this implementation do?

tech-911 commented:
Good morning ma. Help review the new changes and corrections ive made. 
1. removed console logs
2. removed unused components rather than commenting 
3. changed name of image from pp to a more specific name profileAvatar
4. created the UI for profile picture upload and also created the Modal corresponding to the given UI : https://user-images.githubusercontent.com/61203357/201441218-79fe1407-039c-4747-9e7e-aa3a5672e1cf.png

5. added the functionalities of user begin able to upload and delete image uploaded for profile picture

JulianaSau reviewed (changes requested):
Remove the image `packages/main/src/components/top-navigation-bar/assets/language/profileAvatar.png`From the codebase as it's not being used.

I hope you also undid the changes you made before you created the modal

JulianaSau reviewed (changes requested):
Remove the image `packages/main/src/components/top-navigation-bar/assets/language/profileAvatar.png`From the codebase as it's not being used.

I hope you also undid the changes you made before you created the modal

tech-911 commented:
good morning ma yes ma the image packages/main/src/components/top-navigation-bar/assets/language/profileAvatar.png is actually beign used in a state i created . below is a screenshot of the section of the code base to which it was used:

![image](https://user-images.githubusercontent.com/61203357/201460148-787cb078-7a30-4df4-86cb-442ba8a31c3a.png)

1. in the image it was imported then i used it in a useState() hook. 

2. Also for the changes i made earlier yes ma i have reverted them back to normal way it was to be

JulianaSau commented:
![image](https://user-images.githubusercontent.com/49183775/201475199-c0854a54-d92c-4d59-ba95-0c5186f41047.png)
![image](https://user-images.githubusercontent.com/49183775/201475297-934ba062-f793-48f3-996f-f0dc711c0115.png)
The UI already exists on the Zuri Chat page. I'll have to assign another task to you. Close this pr and check the description of your linear task to see the updated task
